src/query/atomWithQuery.ts

```diff
@@ -98,16 +98,17 @@ export function atomWithQuery<
         const unsubscribe = observer.subscribe(listener)
         return unsubscribe
       }
-      return { dataAtom, options }
+      return { dataAtom, observer, options }
     },
     (get, set, action: AtomWithQueryAction) => {
       switch (action.type) {
         case 'refetch': {
-          const { dataAtom, options } = get(queryDataAtom)
+          const { dataAtom, observer, options } = get(queryDataAtom)
           set(dataAtom, new Promise<TData>(() => {})) // infinite pending
-          const queryClient = get(getQueryClientAtom)
-          queryClient.getQueryCache().find(options.queryKey)?.reset()
-          const p: Promise<void> = queryClient.refetchQueries(options.queryKey)
+          const wrappedPromise = async () => {
+            await observer.refetch({ cancelRefetch: true })
```

Comment:
> If `cancelRefetch` is true, then the current request will be cancelled before a new request is made

seems to make sense?

more context: https://react-query.tanstack.com/reference/useQuery#_top

Reply:
@tkdodo curious what you think?

Reply:
seems to make sense from jotai point of view, but I'm not sure how RQ cache works. There might be a reason why it's not default.